File: tools/langgraph-biz-worker/src/langgraph_biz_worker/routes/health.py
"""Health check endpoint (no authentication required)."""

import platform

from fastapi import APIRouter

from .. import __version__
from ..config import settings
from ..external_mode import resolve_external_mode
from ..models import HealthResponse
from ..runtime.agent_capabilities import build_worker_capabilities

router = APIRouter(tags=["health"])

# Active task tracking — imported by query route to register/unregister
active_tasks: set[str] = set()


@router.get("/health", response_model=HealthResponse)
async def health() -> HealthResponse:
    external = resolve_external_mode(settings.external_enabled, settings.worker_token)
    ready = not external.external_enabled or external.external_ready
    return HealthResponse(
        hostname=platform.node(),
        version=__version__,
        active_tasks=len(active_tasks),
        worker_name=settings.worker_name,
        capabilities=build_worker_capabilities(settings.max_agent_nesting_depth),
        status="ok" if ready else "degraded",
        ready=ready,
        mode=external.mode,
        external_enabled=external.external_enabled,
        external_ready=external.external_ready,
        auth_configured=external.auth_configured,
        reasons=list(external.reasons),
    )

File: tools/codex-app-server-worker/src/routes/health.ts
import os from 'node:os'
import { Router } from 'express'
import type { AppConfig } from '../config.js'
import { buildCapabilityManifest, resolveRuntimeReadiness } from '../runtime-capabilities.js'
import type { TaskManager } from '../task-manager.js'
import { APP_VERSION } from '../version.js'
import { resolveExternalModeState } from '../external-mode.js'

export function createHealthRouter(config: AppConfig, manager: TaskManager): Router {
  const router = Router()
  router.get('/health', (_req, res) => {
    const readiness = resolveRuntimeReadiness(config)
    const external = resolveExternalModeState(config)
    if (!manager.isAccepting()) {
      readiness.ready = false
      readiness.reasons.push('APP_SERVER_WORKER_DRAINING')
    }
    res.json({
      status: readiness.ready ? 'ok' : 'degraded',
      ready: readiness.ready,
      reasons: readiness.reasons,
      mode: external.mode,
      external_enabled: external.external_enabled,
      external_ready: external.external_ready,
      auth_configured: external.auth_configured,
      hostname: os.hostname(),
      worker_name: config.workerName,
      version: APP_VERSION,
      runtime_id: config.runtimeId,
      runtime_revision: config.runtimeRevision,
      instance_id: config.instanceId,
      active_tasks: manager.activeCount(),
      queued_tasks: manager.queuedCount(),
      runtime_metrics: manager.runtimeMetrics(),
      cli_version: readiness.cliVersion || null,
      cli_compatible: readiness.cliVersion === '0.144.3',
      state_encryption_configured: Boolean(config.stateEncryptionKey),
    })
  })
  router.get('/api/v1/capabilities', (_req, res) => {
    res.json(buildCapabilityManifest(config, manager))
  })
  return router
}
